controllers/profiles/common/persistence/postgreSQL.go

```diff
@@ -0,0 +1,89 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package persistence
+
+import (
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+
+	operatorapi "github.com/apache/incubator-kie-kogito-serverless-operator/api/v1alpha08"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/controllers/profiles/common/constants"
+)
+
+const (
+	defaultSchemaName   = "default"
+	defaultDatabaseName = "sonataflow"
+)
+
+func ConfigurePostgreSqlEnv(postgresql *operatorapi.PersistencePostgreSql, databaseSchema, databaseNamespace string) []corev1.EnvVar {
+	dataSourcePort := constants.DefaultPostgreSQLPort
+	databaseName := defaultDatabaseName
+	dataSourceURL := postgresql.JdbcUrl
+	if postgresql.ServiceRef != nil {
+		if len(postgresql.ServiceRef.DatabaseSchema) > 0 {
+			databaseSchema = postgresql.ServiceRef.DatabaseSchema
+		}
+		if len(postgresql.ServiceRef.Namespace) > 0 {
+			databaseNamespace = postgresql.ServiceRef.Namespace
+		}
+		if postgresql.ServiceRef.Port != nil {
+			dataSourcePort = *postgresql.ServiceRef.Port
+		}
+		if len(postgresql.ServiceRef.DatabaseName) > 0 {
+			databaseName = postgresql.ServiceRef.DatabaseName
+		}
+		dataSourceURL = "jdbc:" + constants.PersistenceTypePostgreSQL + "://" + postgresql.ServiceRef.Name + "." + databaseNamespace + ":" + strconv.Itoa(dataSourcePort) + "/" + databaseName + "?currentSchema=" + databaseSchema
+	}
+	secretRef := corev1.LocalObjectReference{
+		Name: postgresql.SecretRef.Name,
+	}
+	quarkusDatasourceUsername := "POSTGRESQL_USER"
+	if len(postgresql.SecretRef.UserKey) > 0 {
+		quarkusDatasourceUsername = postgresql.SecretRef.UserKey
+	}
+	quarkusDatasourcePassword := "POSTGRESQL_PASSWORD"
+	if len(postgresql.SecretRef.PasswordKey) > 0 {
+		quarkusDatasourcePassword = postgresql.SecretRef.PasswordKey
+	}
+	return []corev1.EnvVar{
+		{
+			Name: "QUARKUS_DATASOURCE_USERNAME",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key:                  quarkusDatasourceUsername,
+					LocalObjectReference: secretRef,
+				},
+			},
+		},
+		{
+			Name: "QUARKUS_DATASOURCE_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key:                  quarkusDatasourcePassword,
+					LocalObjectReference: secretRef,
+				},
+			},
+		},
+		{
+			Name:  "QUARKUS_DATASOURCE_DB_KIND",
+			Value: constants.PersistenceTypePostgreSQL,
+		},
+		{
+			Name:  "QUARKUS_DATASOURCE_JDBC_URL",
+			Value: dataSourceURL,
+		},
```

Comment:
I'd rather keep all these hardcoded strings in private const.

Reply:
Moved to private constants